### ISSUE TITLE
Stage method stages also conflicted files

### ISSUE THREAD

**Issue by gabriel175** [libgit2/libgit2sharp#1200]
Hi,

When i am working with libgit2sharp i have noticed that once i am using repo.stage("*") it also stages conflicted files (if they exist), which causes the conflicted to be resolved....

I would have been expecting that the repo.stage("*") will have some kind of configuration of "staging all except conflicted files".

Does anyone know how this can be achieved except of reading all the files in the repository and then staging all the files that are not conflicted manually?

Thanks a lot.
Gabi

**Comment by nulltoken** [comment]
Interesting! Just a question, how does git.git react when you run `git add -A` with conflicted files around?

**Comment by carlosmn** [comment]
Staging a conflicted file is precisely how you tell Git that you have resolved the conflict. If you do not mean to say that a conflict has been fixed, you should not stage the file.

**Comment by gabriel175** [comment]
Excatly, so it means that it should be staged manualy with repo.stage(conflictedFile)
But currently when i stage with repo.stage("*") it stages both the conflicted and non conflicted files.
I think that the usage should be somthing like repo.stage("*", StagingStrategy.NonConflicted)

**Comment by ethomson** [comment]
I don't agree.  By default, we should strive to be identical to `git add *`.

**Comment by ethomson** [comment]
Sorry, I was trying to cancel my comment and accidentally closed this.  My bad.

I think that's a reasonable suggestion, though the default really should stay as is.

**Comment by carlosmn** [comment]
>when i stage with repo.stage("") it stages both the conflicted and non conflicted files.

Git/libgit2 don't know that these files are still conflicted. Only the driver of the merge can know this. If you have a use-case where you want to automate creating evil merges, you can get the list of conflicts from the index and remove those from the list of files to be updated in the index.
